### PR TITLE
fix(#2026): Failed to read the 'localStorage' property from 'Window'

### PR DESCRIPTION
**Pull request by Vorobeyko**
## Motivation/Description of the PR

Fix issue  #2026

Applicable helpers:

- [ ] Webdriver
- [x] Puppeteer
- [ ] Nightmare
- [ ] REST
- [ ] Appium
- [ ] Protractor
- [ ] TestCafe

- Description of this PR, which problem it solves
- A link to the corresponding issue (if applicable).

## Type of change

- [ ] Breaking changes
- [ ] New functionality
- [x] Bug fix
- [ ] Documentation changes/updates
- [ ] Hot fix
- [ ] Markdown files fix - not related to source code

## Checklist:

- [x] Tests have been added
- [ ] Documentation has been added (Run `robo docs`)
- [x] Lint checking (Run `npm run lint`)
- [x] Local tests are passed (Run `npm test`)

### CODE REVIEW

File: lib/helper/Puppeteer.js [hunk 1]
```diff
@@ -248,9 +248,13 @@ class Puppeteer extends Helper {
       this.debugSection('Session', 'cleaning cookies and localStorage');
       await this.clearCookie();
     }
-    await this.executeScript('localStorage.clear();').catch((err) => {
-      if (!(err.message.indexOf("Storage is disabled inside 'data:' URLs.") > -1)) throw err;
-    });
+    const currentUrl = await this.grabCurrentUrl()
```
Comment (DavertMik) [inline lib/helper/Puppeteer.js]: Codacy Issue found: [Missing semicolon. (semi)](https://app.codacy.com/app/DavertMik/CodeceptJS/pullRequest?prid=4527839)

**Comment by DavertMik** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/app/CodeceptJS/CodeceptJS/pullRequest?prid=4527839&bid=15244140)

File: lib/helper/Puppeteer.js [hunk 2]
```diff
@@ -248,9 +248,15 @@ class Puppeteer extends Helper {
       this.debugSection('Session', 'cleaning cookies and localStorage');
       await this.clearCookie();
     }
-    await this.executeScript('localStorage.clear();').catch((err) => {
-      if (!(err.message.indexOf("Storage is disabled inside 'data:' URLs.") > -1)) throw err;
-    });
+    const currentUrl = await this.grabCurrentUrl();
+
+    if (currentUrl !== 'about:blank') {
```
Comment (DavertMik) [inline lib/helper/Puppeteer.js]: What if this is a page with `chrome:` ? This is also a special page where everything fails :) 
Maybe we should check that protocol is `http(s):` here?

File: lib/helper/Puppeteer.js [hunk 3]
```diff
@@ -248,9 +248,15 @@ class Puppeteer extends Helper {
...
+      await this.executeScript('localStorage.clear();').catch((err) => {
+        if (!(err.message.indexOf("Storage is disabled inside 'data:' URLs.") > -1)) throw err;
+      });
+    } else {
+      this.debugSection('Session', 'cannot clear localstorage, because current url is "about:blank"');
```
Comment (DavertMik) [inline lib/helper/Puppeteer.js]: Ok, what's the difference between `data:` and `about:` here?
Why do we have 2 ways of handling this?

**Comment by DavertMik** [comment]
Thank you!

File: lib/helper/Puppeteer.js [hunk 4]
```diff
@@ -248,9 +248,13 @@ class Puppeteer extends Helper {
       this.debugSection('Session', 'cleaning cookies and localStorage');
       await this.clearCookie();
     }
-    await this.executeScript('localStorage.clear();').catch((err) => {
-      if (!(err.message.indexOf("Storage is disabled inside 'data:' URLs.") > -1)) throw err;
-    });
+    const currentUrl = await this.grabCurrentUrl();
+
+    if (currentUrl.startsWith('http')) {
```
Comment (elukoyanov) [inline lib/helper/Puppeteer.js]: Looks like will not work for `file`, `ftp` and other protocols.
Hope nobody uses like this )

**Pull request merged**